### PR TITLE
added unit tests

### PR DESCRIPTION
**Pull request by alexsanv**
renamed old one to ...DreddLogs.Tests.Internal.csproj

### CODE REVIEW

File: tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        public void OneTimeSetup()
+        {
+            var testPackage = new TestPackage($"NUnit.OneTimeSetup.DreddLogs.Tests.Internal.dll");
+            _testRunner = TestEngineActivator.CreateInstance().GetRunner(testPackage);
+        }
+
+        [Test]
+        public void SimpleTest()
+        {
+            var testCaseCount = _testRunner.CountTestCases(TestFilter.Empty);
```
Comment (gray-oni) [inline tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs:24]: I would add additional assert that testCaseCount is greater than 0 or expected some particular value (test will become more sensitive but more robust). The main reason to be sure that we will not face a situation when testCaseCount and testCaseNodes = 0 and foreach loop will not be executed.

File: tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+            using (new AssertionScope())
+            {
+                testCaseNodes.Count.Should().Be(testCaseCount);
+
+                foreach (var testCaseNode in testCaseNodes)
+                {
+                    var xmlNode = testCaseNode as XmlNode;
+
+                    var isFailed = xmlNode.SelectSingleNode("./failure") != null;
```
Comment (gray-oni) [inline tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs:37]: How about "is not null" once you use .net5?;)

File: tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                    isFailed.Should().BeTrue();
+
+                    var message = xmlNode.SelectSingleNode("./failure/message").InnerText;
+                    message.Should().Contain("NUnit.OneTimeSetup.DreddLogs.Exceptions.FixtureSetupException : Exception was thrown in fixture setup");
```
Comment (gray-oni) [inline tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs:41]: You can use here useful FluentAssertion capability:
message.Should().Contain("text1")
              .And.Contain("text2);

File: tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs (hunk 2 above)
Comment (alexsanv) [reply tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs:37]: I knew you would notice that :)
Will change it

File: tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs (hunk 1 above)
Comment (alexsanv) [reply tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs:24]: Yeah, makes sense

File: tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs (hunk 3 above)
Comment (alexsanv) [reply tests/NUnit.OneTimeSetup.DreddLogs.Tests/Tests.cs:41]: Okay, thanks for hint

**Review by gray-oni — APPROVED** [review]

**Pull request merged**